### DISCUSSION TITLE
Does SootUp support ```...``` in parameters ?

### DISCUSSION THREAD

**Discussion by Diwu-Yi** [soot-oss/SootUp#571] [Q&A] [answered]
Hi Everyone,

I am wondering if the ... parameter type is currently supported by SootUp.

For example:
<img width="771" alt="Screenshot 2023-03-31 at 4 30 07 PM" src="https://user-images.githubusercontent.com/59159857/229068200-a1897d61-0e8c-442c-a73b-f149eaa05e5b.png">

Is there a way to construct a MethodSignature for the method void addChildren(int index, Node... children)  in class org.jsoup.nodes.Node ? If yes, how it is done, with the ArrayType, I am not sure what to put for ```int dimension``` ?

Thank you in advance for your time.

**Comment by jpstotz** [comment, marked as answer]
AFAIK `Node... children` is just a shortcut in Java Source code for `Node[] children`. 

The Java compiler will always convert it to the array variant and also calls to this method.

**Reply by Diwu-Yi** [reply]
Hi jpstotz,

Thanks for the reply. I get your point. May I further ask, in this case, is it correct to say any integer supplied to the ArrayType will work ?

**Reply by swissiety** [reply]
Here the dimension of the ArrayType is 1 - as its a 1-dimensional array (vs. multidimensional array). Its not refering to the number of elements.

**Reply by JonasKlauke** [reply]
If you wonder how a specific method signature looks like you can simply print all methods of the class
```
    view.getClass(view.getIdentifierFactory().getClassType("Fully_Qualified_Name_of_Class"))
        .ifPresent(javaSootClass -> javaSootClass.getMethods().forEach(System.out::println));
```
view is the view of your project

**Reply by Diwu-Yi** [reply]
> Here the dimension of the ArrayType is 1 - as its a 1-dimensional array (vs. multidimensional array). Its not refering to the number of elements.

Thanks ! This is exactly what I am looking for.
